loopingduptaskproducer/src/main/java/org/duracloud/mill/ltp/dup/LoopingDuplicationTaskProducer.java

```diff
@@ -90,12 +93,27 @@ protected void loadMorselQueueFromSource(Queue<DuplicationMorsel> morselQueue) {
         //generate set of morsels based on duplication policy
         for(String account : this.policyManager.getDuplicationAccounts()){
             DuplicationPolicy policy = this.policyManager.getDuplicationPolicy(account);
-            for(String spaceId : policy.getSpaces()){
-                Set<DuplicationStorePolicy> storePolicies = policy.getDuplicationStorePolicies(spaceId);
-                for(DuplicationStorePolicy storePolicy : storePolicies){
-                    morselQueue.add(new DuplicationMorsel(account, spaceId, null, storePolicy));
+            try {
+                AccountCredentials accountCreds = getCredentialsRepo().getAccountCredentials(account);
+                for(StorageProviderCredentials cred: accountCreds.getProviderCredentials()){
+                    if(cred.isPrimary()){
```

Comment:
This seems to remove the potential for duplication policies to handle duplication using something other than the primary as source. That's not in line with the way the dup policy can be written.

Reply:
That's true.  I'm not sure what I was thinking there, other than the notion that we don't want people accidentally duplicating content from a secondary store to a primary.    Can you think of a case where we would actually want to replicate from a secondary to another secondary? or a secondary to primary?

Reply:
Also it occurs to me now that allowing secondary to primary duplication could potentially set up a continuous loop since duplications generate audit tasks which in turn create duplication messages.

Reply:
The primary use case I had in mind was moving content from one secondary to another secondary if a customer wanted to move, for instance, from SDSC to Glacier. This isn't a critical need, and arguably ensuring we don't accidentally set incorrect policies may be more valuable. 

My main concern is really that the policy editor currently allows for setting up these kinds of secondary to secondary or secondary to primary policies, so if we are moving away from allowing those, the editor shouldn't allow them either.